libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/hybrid/HybridRouterHandler.kt

```diff
@@ -44,79 +45,83 @@ internal sealed class HybridRouterHandler(
     internal class Directions(
         primaryRouter: Router,
         fallbackRouter: Router,
-        logger: Logger
-    ) : HybridRouterHandler(primaryRouter, fallbackRouter, logger) {
+    ) : HybridRouterHandler(primaryRouter, fallbackRouter) {
 
         private inner class PrimaryCallback(
-            private val clientCallback: Router.Callback,
-            private val routeOptions: RouteOptions
-        ) : Router.Callback {
-            override fun onResponse(routes: List<DirectionsRoute>) {
-                clientCallback.onResponse(routes)
+            private val clientCallback: RouterCallback
+        ) : RouterCallback {
+            override fun onRoutesReady(routes: List<DirectionsRoute>) {
+                clientCallback.onRoutesReady(routes)
             }
 
-            override fun onFailure(throwable: Throwable) {
-                logger.w(
+            override fun onFailure(reasons: List<RouterFailure>, routeOptions: RouteOptions) {
+                LoggerProvider.logger.w(
```

Comment:
As mentioned above, we're also logging here. Do we need all this logging "duplication"?

Reply:
I think this one needs to stay - if the primary route fails but the fallback succeeds, developers would never know that primary failed. This log is the only way to get this communicated.